### ISSUE TITLE
Unbiased_lambdamart ndcg is low than original lambdamart

### ISSUE THREAD

**Issue by ubuntuone** [acbull/Unbiased_LambdaMart#3]
I do the follow step:
1. download the generate_dataset
2. python evaluation/scripts/generate_data.py evaluation/click_model/user_browsing_model_0.1_1_4_1.json to generate train and test data
3. run ./lightgbm config="train.conf" and get test data ndcg@10=0.546817
5. build a original lightgbm, version=2.1.1 that same to Unbiased_LambdaMart
6. run ./lightgbm config="train.conf" and get test data ndcg@10=0.556632 
7. Why Unbiased_lambdamart ndcg is low than original lambdamart? The paper say Unbiased_LambdaMart is better than origial?

**Comment by acbull** [comment]
Sorry for the incomplete README and some bugs in the config before...

(1) The original training directory is mistaken. the train_data should be "train_data = ../test_data/letor.user_browsing_model_0_train". I've already changed it.
(2) The data in generate_data are synthetic click data, which is not that meaningful for evaluation. What we really interested in is the result for real data. Therefore, after we get the model output result, we need to:
```
cd ../scripts
python eval.py ../configs/LightGBM_predict_result.txt  #or any other model output.
```
to do evaluation using trec. 

Also by the way, the experiment of the big table in my paper is the one not using regularization. The default regularization term in this Github is 0.5, which will lead to a little bit lower result.

Here is my evaluation output for your reference:

  
![image](https://user-images.githubusercontent.com/31607751/57199082-d76e8000-6f2f-11e9-99dc-d2adeaae2f58.png)

**Comment by ubuntuone** [comment]
Thanks a lot.
I have use eval.py to evaluate test file and get original lightgbm is lower than Unbiased_LambdaMart on NDCG@10.
Now what i confused is I use follow code to cal ndcg is different to trec_eval tools.
`def DCG(label_list):
    dcgsum = 0
    for i in range(len(label_list)):
        dcg = (2 ** label_list[i] - 1) / math.log(i + 2, 2)
        dcgsum += dcg
    return dcgsum
def NDCG(label_list):
    global topK
    dcg = DCG(label_list[0:topK])
    ideal_list = sorted(label_list, reverse=True)
    ideal_dcg = DCG(ideal_list[0:topK])
    if ideal_dcg == 0:
        return 0
    return dcg / ideal_dcg
`

**Comment by acbull** [comment]
For this part, I'm not sure. Maybe you can check: (1) Whether the label data is correct. The groundtruth label is at "generate_dataset/test/test.qrels", instead of the click data test file (that part is only for validation and hyper-parameter tuning); (2) Check the trec's document.

**Comment by ubuntuone** [comment]
Yes, I have checked trec's document and seen ndcg cal is diff. 
Thks!
